core/primitives/src/errors.rs

```diff
@@ -90,7 +90,9 @@ pub enum StorageError {
     /// panic in every place that produces this error.
     /// We can check if db is corrupted by verifying everything in the state trie.
     StorageInconsistentState(String),
-    /// Error from flat storage
+    /// Flat storage error, meaning that it doesn't support some block anymore.
```

Comment:
I think this comment is a bit confusing and doesn't correspond to the name of the error.
I suggest renaming the error to `FlatStorageBlockNotSupported`